### DISCUSSION TITLE
Export Mesh

### DISCUSSION THREAD

**Discussion by Haadis74** [mriscoc/Ender3V2S1#1105] [Q&A]
Hello every 1

Is it somehow possible to export the saved Mesh (modified UBL) to SD card? 

Thanks

**Comment by mriscoc** [comment]
You can use [M420 V](https://marlinfw.org/docs/gcode/M420.html) to get a copy of the current mesh values, then create a[ g-code configuration file](https://github.com/mriscoc/Ender3V2S1/wiki/Configuration-files) to fill the mesh via [M421](https://marlinfw.org/docs/gcode/M421.html)

**Comment by Haadis74** [comment]
I have seen that there is a newer version of the Pro-Firmware than the one I am using. Is there another way to keep the existing Mesh than using M420V? How do you guys do that?

Thank you

**Reply by mriscoc** [reply]
In the UBL versions the mesh is stored in separate EEPROM slots, flashing a new UBL firmware keeps the mesh data.

**Reply by Haadis74** [reply]
Thats good news!! Thank you

**Reply by mriscoc** [reply]
Load your mesh data and save that in other slots for backup.
